src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -734,4 +738,15 @@ CHIP_ERROR OTARequestor::SendNotifyUpdateAppliedRequest(OperationalDeviceProxy &
     return cluster.InvokeCommand(args, this, OnNotifyUpdateAppliedResponse, OnNotifyUpdateAppliedFailure);
 }
 
+// Invoked when the device becomes commissioned
+void OTARequestor::OnCommissioningCompleteRequestor(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    ChipLogProgress(SoftwareUpdate, "Device commissioned, query the default provider");
+
+    // TODO: Should we also send UpdateApplied here?
```

Comment:
No, UpdateApplied doesn't related to commissioning state

Reply:
Yes, but we can't send it before the device is commissioned -- while the application will most likely attempt to send it right after boot and before the device is commissioned

Reply:
Is there anything in the spec that says after a software update has been applied, it must not wipe out certain states such as commissioned? Can this happen? Device is commissioned -> Query/Download/Apply of image succeeds -> upon reboot from apply, the device is no longer commissioned

Reply:
I'm not sure on the answer to this one. @tcarmelveilleux would you happen to know? 

But even if the device does automatically become commissioned after a reboot there is a window where the application starts and -- very likely -- attempts to send UpdateApplied without regard to whether the device has already joined the fabric. 

So sending it here would take care of it. Now the question is: will OnCommissioningCompleteRequestor() still be called if the device re-joins the fabric without being explicitly commissioned?

Reply:
> But even if the device does automatically become commissioned after a reboot there is a window where the application starts and -- very likely -- attempts to send UpdateApplied without regard to whether the device has already joined the fabric.

Perhaps in cases like this, it is the application's responsibility to wait before calling to notify. We could even put it in the header to say that is the expectation. Agree with @tcarmelveilleux here that we should only be handling QueryImage after commission, not for UpdateApplied.

Reply:
OK, closing the conversation if there are no other comments.